tests/python/unittest/common.py

```diff
@@ -308,6 +309,7 @@ def teardown():
     It waits for all operations in one file to finish before carrying on the next.
     """
     mx.nd.waitall()
+    gc.collect()
```

Comment:
Just curious. Did this actually help?

Reply:
Actually not sure. I didn't know if there was a way to identify whether it was helpful or not. In a nutshell I still ran out of memory for 480GB machine.

Reply:
Have played with teardown multiple times before.
Learnt only 1 thing : MXNet Memory Management would be a great project :p